### DISCUSSION TITLE
Laptop upgrade

### DISCUSSION THREAD

**Discussion by 0xSecurious** [Cyfrin/foundry-full-course-cu#1978] [General]
Hi! I am thinking about buying a new laptop, since my old one is literally dying. It has 4gb of ram and an old i3 processor. I'd like to learn web3 development, maybe some security in the future. Since i am a begginer, it is possible for me to change into some other branch of development. Would like to go into Macbook territory, but can't decide on the model. What would be your best recommendations for my situation? Maybe more of a learning machine than a workhorse

**Comment by EngrPips** [comment]
Honestly, I think any Macbook you get is fine. Just get the more recent ones. MacBook 2018 is fine (air or pro).

**Reply by 0xSecurious** [reply]
I simply don't know much about the requirements for writing solidity code, smart contracts etc. From what i found, better get 16gb ram and the latter doesn't really matter much i guess. As for operating a node, i guess you need some beefy stuff

Thanks for your reply!

**Reply by EngrPips** [reply]
You are welcome, friend. I don't know much about computers myself as well. I use a MacBook Pro 2019, and it works fine, at least.
